### PR TITLE
Adds saveNextFederationCreationBlockHeight test

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds saveNextFederationCreationBlockHeight test

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -1162,6 +1162,30 @@ void saveActiveFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNot
 
     }
 
+    @Test
+    void saveNextFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNotNullAndRSKIP186IsActive() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1166]: same as [here](https://github.com/rsksmart/rskj/pull/2557/files#r1669155796)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -1162,6 +1162,30 @@ void saveActiveFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNot
...
+        // Arrange
+        StorageAccessor storageAccessor = new InMemoryStorage();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1168]: same as [here](https://github.com/rsksmart/rskj/pull/2557/files#r1669147235)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -1162,6 +1162,30 @@ void saveActiveFederationCreationBlockHeight_shouldSaveToStorage_whenHeightIsNot
...
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        long expectedFederationCreationBlockHeight = 100;
+        federationStorageProvider.setNextFederationCreationBlockHeight(expectedFederationCreationBlockHeight);
+
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1175]: same as [here](https://github.com/rsksmart/rskj/pull/2557/files#r1669146738)

**Review by wilmerrootstock — APPROVED** [review]
LGTM!

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

**Review by julia-zack — APPROVED** [review]

**Pull request closed without merging**